noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/components/tail_to_public_output_composer/meter_gas_used.nr

```diff
@@ -36,15 +40,15 @@ fn meter_gas_used(data: PublicAccumulatedData) -> Gas {
     metered_da_bytes += unencrypted_log_preimages_length as u32;
     metered_l2_gas += unencrypted_log_preimages_length as u32 * L2_GAS_PER_LOG_BYTE;
 
-    metered_l2_gas += array_length(data.public_call_stack) * FIXED_AVM_STARTUP_L2_GAS;
+    metered_l2_gas += array_length(data.public_call_requests) * FIXED_AVM_STARTUP_L2_GAS;
 
     Gas::new(metered_da_bytes * DA_GAS_PER_BYTE, metered_l2_gas)
 }
 
-pub fn meter_gas_used_non_revertible(data: PublicAccumulatedData) -> Gas {
+pub fn meter_gas_used_non_revertible(data: PrivateToPublicAccumulatedData) -> Gas {
     meter_gas_used(data) + Gas::tx_overhead()
 }
 
-pub fn meter_gas_used_revertible(data: PublicAccumulatedData, teardown_gas: Gas) -> Gas {
-    meter_gas_used(data) + Gas::new(teardown_gas.da_gas, teardown_gas.l2_gas)
```

Comment:
Don't add the `teardown_gas` to revertible `gas_used` in private_tail_to_public.

It is used to compute the transaction fee, and should be added when actually computing the transaction fee. (I think) it was added here to make the code for computing fee slightly simpler - adding gas used from both non-revertible and revertible. But there were code in some places that subtracted the teardown gas to get the actual gas used, which seemed more confusing.